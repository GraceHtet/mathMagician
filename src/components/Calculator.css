.cal-container {
  width: 30vw;
  background: navajowhite;
  color: #121212;
  font-size: 18px;
  line-height: 3;
  border-radius: 10px;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr;
  overflow: hidden;
}

.display {
  text-align: right;
  padding-right: 10%;
  font-weight: bold;
  font-size: 24px;
  grid-area: 1/ 1 / span 1 / span 4;
}

.btn {
  border: none;
  background: rgba(0, 0, 0, 0.8);
  color: #fff;
  cursor: pointer;
}

.btn:nth-of-type(17) {
  grid-area: 6/1 / span 1 / span 2;
}

.btn:hover {
  background: rgba(0, 0, 0, 0.5);
  border: 2px solid #aaa;
}
