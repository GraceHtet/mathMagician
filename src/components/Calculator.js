import './Calculator.css';

const Calculator = () => {
  const numSigns = [
    'AC',
    '+/-',
    '%',
    '/',
    7,
    8,
    9,
    '*',
    4,
    5,
    6,
    '-',
    1,
    2,
    3,
    '+',
    0,
    '.',
    '=',
  ];
  return (
    <div className="cal-container">
      <div className="display">0</div>
      {numSigns.map((num) => (
        <button type="button" className="btn" key={num}>
          {num}
        </button>
      ))}
    </div>
  );
};

export default Calculator;
