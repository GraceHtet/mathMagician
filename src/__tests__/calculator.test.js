import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Calculator from '../components/Calculator';

const numSigns = [
  'AC',
  '+/-',
  '%',
  'รท',
  '7',
  '8',
  '9',
  'x',
  '4',
  '5',
  '6',
  '-',
  '1',
  '2',
  '3',
  '+',
  '0',
  '.',
  '=',
];

describe('Calculator', () => {
  test('renders Display component', () => {
    render(<Calculator />);
    expect(screen.getByTestId('display')).toMatchSnapshot();
  });

  numSigns.forEach((numsign) => {
    test(`show ${numsign}`, () => {
      render(<Calculator />);
      expect(screen.getByText(numsign)).toMatchSnapshot();
    });
  });

  test('the reult of (1+3-10.569x7รท9%11=) should equal 5.10922 ', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('-'));
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('.'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('6'));
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('รท'));
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('%'));
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('='));
    fireEvent.click(screen.getByText('+/-'));
    expect(screen.getByTestId('display').textContent).toBe('5.10922');
  });

  test('the reult of (3.56 +/-) should equal -3.56 ', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('.'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('6'));
    fireEvent.click(screen.getByText('+/-'));
    expect(screen.getByTestId('display').textContent).toBe('-3.56');
  });
});
