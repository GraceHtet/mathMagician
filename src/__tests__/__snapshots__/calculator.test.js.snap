// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders Display component 1`] = `
<div
  class="display"
  data-testid="display"
>
  <p />
</div>
`;

exports[`Calculator show % 1`] = `
<button
  class="btn"
  type="button"
>
  %
</button>
`;

exports[`Calculator show + 1`] = `
<button
  class="btn"
  type="button"
>
  +
</button>
`;

exports[`Calculator show +/- 1`] = `
<button
  class="btn"
  type="button"
>
  +/-
</button>
`;

exports[`Calculator show . 1`] = `
<button
  class="btn"
  type="button"
>
  .
</button>
`;

exports[`Calculator show = 1`] = `
<button
  class="btn"
  type="button"
>
  =
</button>
`;

exports[`Calculator show - 1`] = `
<button
  class="btn"
  type="button"
>
  -
</button>
`;

exports[`Calculator show 0 1`] = `
<button
  class="btn"
  type="button"
>
  0
</button>
`;

exports[`Calculator show 1 1`] = `
<button
  class="btn"
  type="button"
>
  1
</button>
`;

exports[`Calculator show 2 1`] = `
<button
  class="btn"
  type="button"
>
  2
</button>
`;

exports[`Calculator show 3 1`] = `
<button
  class="btn"
  type="button"
>
  3
</button>
`;

exports[`Calculator show 4 1`] = `
<button
  class="btn"
  type="button"
>
  4
</button>
`;

exports[`Calculator show 5 1`] = `
<button
  class="btn"
  type="button"
>
  5
</button>
`;

exports[`Calculator show 6 1`] = `
<button
  class="btn"
  type="button"
>
  6
</button>
`;

exports[`Calculator show 7 1`] = `
<button
  class="btn"
  type="button"
>
  7
</button>
`;

exports[`Calculator show 8 1`] = `
<button
  class="btn"
  type="button"
>
  8
</button>
`;

exports[`Calculator show 9 1`] = `
<button
  class="btn"
  type="button"
>
  9
</button>
`;

exports[`Calculator show AC 1`] = `
<button
  class="btn"
  type="button"
>
  AC
</button>
`;

exports[`Calculator show x 1`] = `
<button
  class="btn"
  type="button"
>
  x
</button>
`;

exports[`Calculator show รท 1`] = `
<button
  class="btn"
  type="button"
>
  รท
</button>
`;
